app/src/swap/handle_sign_transaction.c

```diff
@@ -0,0 +1,187 @@
+/*******************************************************************************
+ *   (c) 2016 Ledger
+ *   (c) 2018 - 2023 Zondax AG
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#include "bignum.h"
+#include "crypto.h"
+#include "lib_standard_app/swap_lib_calls.h"
+#include "parser.h"
+#include "swap.h"
+#include "swap_utils.h"
+#include "zxformat.h"
+
+#define SWAP_EXPECTED_ITEMS 3
+
+swap_globals_t G_swap_state;
+
+// Save the BSS address where we will write the return value when finished
+static uint8_t *G_swap_sign_return_value_address;
+
+////////////////////////////////////////////////////////////////
+
+bool copy_transaction_parameters(create_transaction_parameters_t *sign_transaction_params) {
+    if (sign_transaction_params == NULL) {
+        return false;
+    }
+
+    // First copy parameters to stack, and then to global data.
+    // We need this "trick" as the input data position can overlap with globals
+    char destination_address[65] = {0};
+    uint8_t amount[16] = {0};
+    uint8_t amount_length = {0};
+    uint8_t fees[8] = {0};
```

Comment:
I find it easier to read when magic numbers are replaced by macros. It's easier to check that the size is correctly checked and operations are in bounds. But you can consider this as informational.
On a more positive side, this code is really well documented making it easy to read!

Reply:
Done